lib/system/sysio.nim

```diff
@@ -97,23 +97,28 @@ else:
 const
   BufSize = 4000
 
+proc close*(f: File) = discard c_fclose(f)
+proc readChar*(f: File): char = result = cast[char](c_fgetc(f))
```

Comment:
type conversion will throw, no? I don't think readChar did, as a side effect of the cast happening in c due to the invalid signature.. which is a little unfortunate for values like `EOF`